### PR TITLE
(chores) ci: added support for generating summaries from the incremen…

### PR DESCRIPTION
**Pull request by LostArtist**
…tal test script

# Description

# Target

- [ ] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [ ] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [ ] I checked that each commit in the pull request has a meaningful subject line and body.

- [ ] I have run `mvn clean install -DskipTests` locally and I have committed all auto-generated changes

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by orpiske** [comment]
This is a WIP. There's a few improvements that will follow this ... just bear w/ us because GH scripts are a bit hard to test.

File: .github/actions/incremental-build/parse_errors.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+LOG_FILE=$1
+
+OUTPUT_FILE="error_messages.txt"
+
+echo "| Failed Test |
```
Comment (orpiske) [inline .github/actions/incremental-build/parse_errors.sh:23]: 1. Missing the double quotes (") closing the first line of text. 
2. Missing saving the table header to the summary file (```> "$GITHUB_STEP_SUMMARY"```)

File: .github/actions/incremental-build/parse_errors.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+| --- |" > "$GITHUB_STEP_SUMMARY"
```
Comment (orpiske) [inline .github/actions/incremental-build/parse_errors.sh:24]: Needs to be modified, for appending after fixing the items above.

Comment (orpiske) [reply .github/actions/incremental-build/parse_errors.sh:24]: For both lines: they need not to be written if there is no errors on the log file.

**Review by orpiske — CHANGES_REQUESTED** [review]
There's a few things to be fixed before we can merge.

**Review by orpiske — APPROVED** [review]

**Pull request merged**